internal/filesystem/ipfs/client.go

```diff
@@ -0,0 +1,87 @@
+package ipfs
+
+import (
+	"context"
+	"errors"
+	"net"
+	"net/http"
+	"time"
+
+	httpapi "github.com/ipfs/go-ipfs-http-client"
```

Comment:
>The IPFS client library was deprecated again.

  Correction, `ipfs/go-ipfs-api` is the one being deprecated, _not_ this one (yet?).

We'll have to re-evaluate all the different client pkgs we use anyhow since there's multiple in play here.
Maybe things can stay as they are for now. But it would be nice to be able to consolidate.

Reply:
>kubo RPC clients

Thanks for the heads up.
We'll probably move forward with the old dependency first since that is known to work in our manual testing.

Since the new pkg is currently throwing an error somewhere, I'll have to track down what's going on there and ideally add automated tests for the `fs.FS` implementations that pass with both pkgs before migrating to the new one.
That's going to require setting up a fresh IPFS instance that we can connect to via rpc, populating it with some test data, then actually testing the `fs.FS` implementations against it.

>If there's an API you want access to sooner in

There's been requests for mounting the IPFS `/files` API, which I'll likely be working on soon.

>file an issue or PR

There's some pre-existing work that may be usable as a base, but has since been abandoned.
https://github.com/ipfs/kubo/pull/6081
https://github.com/ipfs/interface-go-ipfs-core/pull/54

I'm thinking it might be easier to hack something together outside of the IPFS repos that satisfies our needs, rather than waiting on upstream to provide a sanctioned and general API.

Reply:
Done in: 839db7c0eca015d1c92359ac4e5f9f8462623b06